src/lib/containers/entity/annotations/SchemaDrivenAnnotationEditor.tsx

```diff
@@ -147,19 +150,22 @@ export const SchemaDrivenAnnotationEditor: React.FunctionComponent<SchemaDrivenA
 
   const isLoading = isLoadingBinding || isLoadingSchema
 
-  const mutation = useUpdateViaJson(
-    entityId!,
-    { ...dropNullishArrayValues(formData ?? {}), ...entityJson },
-    {
-      onSuccess: () => {
-        onSuccess()
-      },
-      onError: error => {
-        setSubmissionError(error)
-        setShowSubmissionError(true)
-      },
+  const mutation = useUpdateViaJson({
+    onSuccess: () => {
+      onSuccess()
     },
-  )
+    onError: error => {
+      setSubmissionError(error)
+      setShowSubmissionError(true)
+    },
+  })
+
+  function submitChangedEntity() {
+    mutation.mutate({
+      ...dropNullishArrayValues(formData ?? {}),
+      ...entityJson,
+    } as EntityJson)
+  }
 
```

Comment:
Instead of calling  `mutation.mutate()`, call `submitChangedEntity`	which calls `mutation.mutate(newJson)`